ichnaea/api/locate/tests/base.py

```diff
@@ -561,67 +595,85 @@ def test_cell_invalid_lac(self, app, data_queues):
         res = self._call(app, body=query, status=self.not_found.code)
         self.check_response(data_queues, res, "not_found")
 
-    def test_cell_lte_radio(self, app, session, stats):
+    def test_cell_lte_radio(self, app, session, metricsmock):
         cell = CellShardFactory(radio=Radio.lte)
         session.flush()
 
         query = self.model_query(cells=[cell])
         res = self._call(app, body=query)
         self.check_model_response(res, cell)
-        stats.check(
-            counter=[
-                (self.metric_type + ".request", [self.metric_path, "key:test"]),
-                ("request", [self.metric_path, "method:post", "status:200"]),
-            ]
+        assert metricsmock.has_record(
+            "incr",
+            self.metric_type + ".request",
+            value=1,
+            tags=[self.metric_path, "key:test"],
+        )
+        assert metricsmock.has_record(
```

Comment:
I'm tempted to tweak markus to have `.has_incr`, `.has_gauge`, `.has_timing`, and `.has_histogram`. It reduces a lot of these by at least a line without losing any clarity.

@jwhitlock What do you think about that?

Reply:
Oh, that's brilliant! I'll look at what other mock assertions make sense to have. It's very much like assertions on function calls.